boilerplate/fullstack/remix/app/lib/superTokensHelpers.ts

```diff
@@ -0,0 +1,121 @@
+    return new PreParsedRequest({
+        cookies: getCookieFromRequest(request),
+        url: request.url as string,
+        method: request.method as HTTPMethod,
+        query: getQueryFromRequest(request),
+        headers: request.headers,
+        getFormBody: async () => {
+            return await request.formData();
+        },
+        getJSONBody: async () => {
+            return await request.json();
+        },
+    });
+}
+
+export async function getSessionDetails(
+    request: Request,
+    options?: VerifySessionOptions,
+    userContext?: Record<string, unknown>
+): Promise<{
+    session: SessionContainer | undefined;
+    hasToken: boolean;
+    hasInvalidClaims: boolean;
+    nextResponse?: Response;
+}> {
+    console.log("Getting session details...");
+
+    const baseRequest = createPreParsedRequest(request);
+    console.log("Pre-parsed request created.");
+
+    const baseResponse = new CollectingResponse();
+    console.log("Collecting response object created.");
+
+    // Possible introp issue.
+    const recipe = (SessionRecipe as any).default.instance;
+    console.log("Session recipe instance obtained.");
+
+    const tokenTransferMethod = recipe.config.getTokenTransferMethod({
+        req: baseRequest,
+        forCreateNewSession: false,
+        userContext,
+    });
+    console.log("Token transfer method determined:", tokenTransferMethod);
+
+    const transferMethods = tokenTransferMethod === "any" ? availableTokenTransferMethods : [tokenTransferMethod];
+    console.log("Available token transfer methods:", transferMethods);
+
+    const hasToken = transferMethods.some((transferMethod) => {
+        const token = getToken(baseRequest, "access", transferMethod);
+        if (!token) {
+            console.log("Token not found for transfer method:", transferMethod);
+            return false;
+        }
+        try {
+            parseJWTWithoutSignatureVerification(token);
+            console.log("Token parsed successfully.");
+            return true;
+        } catch {
+            console.log("Failed to parse token:", token);
+            return false;
+        }
+    });
+    console.log("Does the user have a token?", hasToken);
+
+    try {
+        const session = await Session.getSession(baseRequest, baseResponse, options, userContext);
+        console.log("Session obtained:", session);
+        return {
+            session,
+            hasInvalidClaims: false,
+            hasToken,
+        };
+    } catch (err) {
+        console.error("Error while getting session:", err);
+        if (Session.Error.isErrorFromSuperTokens(err)) {
+            console.log("SuperTokens error detected.");
+            return {
+                hasToken,
+                hasInvalidClaims: err.type === Session.Error.INVALID_CLAIMS,
+                session: undefined,
+                nextResponse: new Response("Authentication required", {
```

Comment:
if we are going to send a 403, you should send back a JSON with the claim error. The claim error is in the Error object somewhere. Inspect it, or see in our SDK

Reply:
I discussed this with @anku255 briefly, and I got some pointers. Now, I'm using this method in two scenarios after the latest changes - as a loader (the SSR part) in `routes/_index.tsx` and as an API response in `routes/supertokens.sessionsinfo.$.tsx`. 

As @anku255 explained, throwing this: `new SessionError({ type: SessionError.INVALID_CLAIMS, message: "Invalid claims!" });` would only work if our error handler catches it. Any advice on how to go about that?